packages/jellyfish-api-core/__tests__/category/vault/estimateVault.test.ts

```diff
@@ -0,0 +1,165 @@
+
+describe('Vault estimateVault', () => {
+  const tGroup = TestingGroup.create(1, i => new MasterNodeRegTestContainer())
+
+  async function setup (tGroup: TestingGroup): Promise<void> {
+    // token setup
+    const collateralAddress = await tGroup.get(0).container.getNewAddress()
+    await tGroup.get(0).token.dfi({ address: collateralAddress, amount: 30000 })
+    await tGroup.get(0).generate(1)
+    // Setup collateral token
+    await tGroup.get(0).token.create({ symbol: 'BTC', collateralAddress })
+    await tGroup.get(0).generate(1)
+    await tGroup.get(0).token.mint({ symbol: 'BTC', amount: 20000 })
+    await tGroup.get(0).generate(1)
+    // Setup non collateral token
+    await tGroup.get(0).token.create({ symbol: 'DOGE', collateralAddress })
+    await tGroup.get(0).generate(1)
+    await tGroup.get(0).token.mint({ symbol: 'DOGE', amount: 50000 })
+    await tGroup.get(0).generate(1)
+
+    // oracle setup
+    const addr = await tGroup.get(0).generateAddress()
+    const priceFeeds = [
+      { token: 'DFI', currency: 'USD' },
+      { token: 'BTC', currency: 'USD' },
+      { token: 'DOGE', currency: 'USD' },
+      { token: 'TSLA', currency: 'USD' },
+      { token: 'MSFT', currency: 'USD' }
+    ]
+    const oracleId = await tGroup.get(0).rpc.oracle.appointOracle(addr, priceFeeds, { weightage: 1 })
+    await tGroup.get(0).generate(1)
+    const oracleTickTimestamp = Math.floor(new Date().getTime() / 1000)
+    await tGroup.get(0).rpc.oracle.setOracleData(oracleId, oracleTickTimestamp, {
+      prices: [
+        { tokenAmount: '1@DFI', currency: 'USD' },
+        { tokenAmount: '0.01@DOGE', currency: 'USD' },
+        { tokenAmount: '10000@BTC', currency: 'USD' },
+        { tokenAmount: '2@TSLA', currency: 'USD' },
+        { tokenAmount: '5@MSFT', currency: 'USD' }
+      ]
+    })
+    await tGroup.get(0).generate(1)
+
+    // collateral token
+    await tGroup.get(0).rpc.loan.setCollateralToken({
+      token: 'DFI',
+      factor: new BigNumber(1),
+      fixedIntervalPriceId: 'DFI/USD'
+    })
+    await tGroup.get(0).generate(1)
+
+    // collateral token
+    await tGroup.get(0).rpc.loan.setCollateralToken({
+      token: 'BTC',
+      factor: new BigNumber(0.5),
+      fixedIntervalPriceId: 'BTC/USD'
+    })
+    await tGroup.get(0).generate(1)
+
+    // loan token
+    await tGroup.get(0).rpc.loan.setLoanToken({
+      symbol: 'TSLA',
+      fixedIntervalPriceId: 'TSLA/USD'
+    })
+    await tGroup.get(0).generate(1)
+    await tGroup.get(0).rpc.loan.setLoanToken({
+      symbol: 'MSFT',
+      fixedIntervalPriceId: 'MSFT/USD'
+    })
+    await tGroup.get(0).generate(1)
+
+    // loan scheme set up
+    await tGroup.get(0).rpc.loan.createLoanScheme({
+      minColRatio: 150,
+      interestRate: new BigNumber(3),
+      id: 'scheme'
+    })
+    await tGroup.get(0).generate(1)
+
+    await tGroup.get(0).rpc.oracle.setOracleData(oracleId, Math.floor(new Date().getTime() / 1000), {
```

Comment:
You may remove this setOracleData. Because you have written 2@TSLA and 5@MSFT here
https://github.com/DeFiCh/jellyfish/pull/1354/files#diff-6da836d6c16b6122852b6da75a6a3761acede7daf6fcf3f8c65fe52776c64840R43